rust/signed_doc/src/payload/json.rs

```diff
@@ -0,0 +1,41 @@
+//! JSON Content
+use super::Content;
+use crate::metadata::ContentEncoding;
+
+/// JSON encoded content
+pub type Json = Content<serde_json::Value>;
+
+impl Default for Json {
+    fn default() -> Self {
+        serde_json::Value::Object(serde_json::Map::new()).into()
+    }
+}
+
+impl TryFrom<&[u8]> for Content<serde_json::Value> {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
+        serde_json::from_slice(value)
+            .map_err(|e| anyhow::anyhow!("Failed to parse any JSON content: {e}"))
+    }
+}
+
+impl TryFrom<(&Vec<u8>, Option<ContentEncoding>)> for Content<serde_json::Value> {
```

Comment:
It make sense to make this implementation generic:
```rust
impl<T: TryFrom&[u8]>> TryFrom<(&Vec<u8>, Option<ContentEncoding>)> for Content<T> {}
```

Reply:
This is mainly done to provide the content encoding error, which is later recorded as a validation error. I'll add proper errors so that this cleaner.